core/src/main/java/io/kestra/core/models/flows/FlowWithSource.java

```diff
@@ -40,22 +39,11 @@ public Flow toFlow() {
             .build();
     }
 
-    public String getSource() {
-        String source = this.source;
-
-        // previously, we insert source on database keeping default value (like deleted, ...)
-        // if the previous serialization is the same as actual one, we use a clean version removing them
-        Flow flow = toFlow();
-        source = FlowService.generateSource(flow, source);
-
-        // same here but with version that don't make any sense on the source code, so removing it
-        return cleanupSource(source);
-    }
-
     private static String cleanupSource(String source) {
         return source.replaceFirst("(?m)^revision: \\d+\n?","");
     }
 
+    // TODO check with Yann if we can remove this.source.equals(cleanupSource(flowSource)) here
     public boolean equals(Flow flow, String flowSource) {
```

Comment:
This may need to be tested, but I believe we need to keep it, as however flow that just have there comments/syntax edited may not be considered modified

Reply:
OK, reading the code source it matters where it is used.